src/library/filescanner_ffmpeg.c

```diff
@@ -295,44 +310,64 @@ extract_metadata_core(struct media_file_info *mfi, AVDictionary *md, const struc
     DPRINTF(E_DBG, L_SCAN, " -> %s = %s\n", mdt->key, mdt->value);
 #endif
 
-  mdcount = 0;
-
-  /* Extract actual metadata */
-  for (i = 0; md_map[i].key != NULL; i++)
+  mdcount = 0;    
+  
+  /* Cache this search once to avoid doing it per metadata key */
+  lyricsindex = match_metadata("lyrics", md_map);
```

Comment:
If I understand this correctly it just saves cycles in the case where a file has lyrics in many different languages? I'm not sure that's worth optimizing. Multiple languages will cause another issue, which is that we aren't necessarily picking the one the user wants. Basically I think that use case is so exotic that we shouldn't cater for it.

Reply:
No: this finds the index of Owntone's "lyrics" key in the static md_map (which is static and fixed at compile time). So it's done once (currently per file, could be moved out to be done per run) instead of being done once per metadata found in the file.

The idea is:
```python
   for (each metadata found in file):
      if metadataKey starts with "lyrics":
         lyricsIndex = find the index in md_map for the key "lyrics"   << This is static and knowable at compile time
         save the lyrics in the metadata array at index lyricsIndex
     else:
        index = find the index in md_map for the metadataKey
        save the metadata in the metadata array at index index 
```

Here, the lyricsIndex is known at compile time, so there's no point in searching it for each lyrics (and logically even for any file). Ideally, it could even be saved once for all when starting the whole parsing process, but I haven't done that yet.

Reply:
Moved to a static index so it's only done once per runtime.

Reply:
Maybe I didn't explain myself well enough. I did understand that it saved the fixed index. Here's examples to show you what I mean:

- 5 metadata, none of them lyrics: 6 iterations (one to find the fixed index of lyrics + 5 x match_metadata)
- 5 metadata , one of them lyrics:  5 iterations (one to find the fixed index of lyrics + 4 x match_metadata)
- 5 metadata , two of them lyrics:  5 iterations (one to find the fixed index of lyrics + 3 x match_metadata)

If you instead just set the key to "lyrics" for any variation of "lyrics-" and then proceed as normal, all the above will require 5 iterations and the code will be more simple.

Reply:
Ok. That's right. I've changed it so now it's done once for all, so it'll only cost one search per launch of the program and none afterward. Should be better now.